modules/exploits/windows/smb/ms17_010_eternalblue_win8.rb

```diff
@@ -0,0 +1,951 @@
+    super(
+      update_info(
+        info,
+        'Name' => 'MS17-010 EternalBlue SMB Remote Windows Kernel Pool Corruption for Win8+',
+        'Description' => %q{
+          EternalBlue exploit for Windows 8, Windows 10, and 2012 by sleepya
+          The exploit might FAIL and CRASH a target system (depended on what is overwritten)
+          The exploit support only x64 target
+
+          Tested on:
+          - Windows 2012 R2 x64
+          - Windows 8.1 x64
+          - Windows 10 Pro Build 10240 x64
+          - Windows 10 Enterprise Evaluation Build 10586 x64
+
+
+          Default Windows 8 and later installation without additional service info:
+          - anonymous is not allowed to access any share (including IPC$)
+          - More info: https://support.microsoft.com/en-us/help/3034016/ipc-share-and-null-session-behavior-in-windows
+          - tcp port 445 is filtered by firewall
+
+
+          Reference:
+          - http://blogs.360.cn/360safe/2017/04/17/nsa-eternalblue-smb/
+          - "Bypassing Windows 10 kernel ASLR (remote) by Stefan Le Berre" https://drive.google.com/file/d/0B3P18M-shbwrNWZTa181ZWRCclk/edit
+
+
+          Exploit info:
+          - If you do not know how exploit for Windows 7/2008 work. Please read my exploit for Windows 7/2008 at
+          https://gist.github.com/worawit/bd04bad3cd231474763b873df081c09a because the trick for exploit is almost the same
+          - The exploit use heap of HAL for placing fake struct (address 0xffffffffffd00e00) and shellcode (address 0xffffffffffd01000).
+          On Windows 8 and Wndows 2012, the NX bit is set on this memory page. Need to disable it before controlling RIP.
+          - The exploit is likely to crash a target when it failed
+          - The overflow is happened on nonpaged pool so we need to massage target nonpaged pool.
+          - If exploit failed but target does not crash, try increasing 'GroomAllocations' value (at least 5)
+          - See the code and comment for exploit detail.
+
+
+          Disable NX method:
+          - The idea is from "Bypassing Windows 10 kernel ASLR (remote) by Stefan Le Berre" (see link in reference)
+          - The exploit is also the same but we need to trigger bug twice
+          - First trigger, set MDL.MappedSystemVa to target pte address
+          - Write '\x00' to disable the NX flag
+          - Second trigger, do the same as Windows 7 exploit
+          - From my test, if exploit disable NX successfully, I always get code execution
+        },
+        'Author' =>
+          [
+            'Equation Group',  # OG research and exploit
+            'Shadow Brokers',  # Hack and dump
+            'sleepya',         # Research and PoC
+            'wvu',             # Babby's first external module
+            'agalway-r7',      # External python module to internal ruby module (sorry wvu)
+            'cdelafuente-r7',  # ruby_smb wizard
+            'cdelafuente-r7'   # kernel debugging wizard
```

Comment:
```suggestion
            'Equation Group', # OG research and exploit
            'Shadow Brokers', # Hack and dump
            'sleepya', # Research and PoC
            'wvu', # Babby's first external module
            'agalway-r7', # External python module to internal ruby module (sorry wvu)
            'cdelafuente-r7' # ruby_smb and kernel debugging wizard
```

[PEP 8 no longer applies.](https://www.python.org/dev/peps/pep-0008/#inline-comments) Feel free to align or not. I don't think @adfoster-r7 likes to.